test/tools/d_do_test.d

```diff
@@ -103,12 +103,98 @@ struct EnvData
     string ccompiler;
     string model;
     string required_args;
+    string cxxCompatFlags;      /// C++ interop flags passed to $(compiler)
+    string[] picFlag;           /// Compiler flag for PIC (if requested from environment)
     bool dobjc;
     bool coverage_build;
     bool autoUpdate;
+    bool printRuntime;          /// Print time spent on a single test
     bool usingMicrosoftCompiler;
 }
 
+/++
+Creates a new EnvData instance based on the current environment.
+Other code should not read from the environment.
+
+Returns: an initialized EnvData instance
+++/
+immutable(EnvData) processEnvironment()
```

Comment:
Not sure why you would go with `immutable` over `const` here (but it doesn't matter much, just curious).

Reply:
My goal is to import `d_do_test.d` from `run.d` instead of spawning a new process for every test, hence it needs to be structured in a way that can be called for every tests without doing the same stuff over and over again.
Initially wanted to make `envData` an immutable global (it's just a wrapper over `environment` after all). But that obviously doesn't work because one cannot `SilentQuit` from a `shared static this()`.
This is still valuable because immutable signifies that it can never change after it's creation.